src/components/album/AlbumsList.js

```diff
@@ -22,41 +24,60 @@ import {
 
 const AlbumsList = () => {
   const [albums, setAlbums] = useState([]); // for albums
-  const [limit, setLimit] = useState(1); // for pagination
+  const [totalPages, setTotalPages] = useState(1); // Track total pages for this search
+  const limit = 12;
+  // const [limit, setLimit] = useState(12); // for pagination
   const [searchType, setSearchType] = useState('albumName'); //default value for select input
   const [searchTerm, setSearchTerm] = useState(''); // for search input filed value
   const [message, setMessage] = useState(''); // for not found message
   const [errorMessage, setErrorMessage] = useState(''); // for error message
   const [wishListId, setWishListId] = useState();
+  const [currentPage, setCurrentPage] = useState(1); // Track current page
+  // const [lastSearchTerm, setLastSearchTerm] = useState('');
 
   //make an API call with search values to backend and return the result
-  const fetchAlbums = async (searchType, searchTerm, limit) => {
-    try {
-      const response = await axios.get(
-        `${process.env.REACT_APP_API_BASE_PATH}/albums/filter`,
-        {
-          params: {
-            limit,
-            [searchType]: searchTerm,
-          },
+  const fetchAlbums = useCallback(
+    async (searchType, searchTerm) => {
+      try {
+        //const offset = (currentPage - 1) * limit;
+        const offset = (searchTerm === '' ? currentPage - 1 : 0) * limit;
+
+        //for testing
+        // console.log(currentPage - 1 + ' * ' + limit + ' = ' + offset);
+        // console.log('searchType:' + searchType + ' ; searchTerm:' + searchTerm);
+
+        const response = await axios.get(
+          `${process.env.REACT_APP_API_BASE_PATH}/albums/filter`,
```

Comment:
We should prefer in this repo to use `axiosInstance` from the `src/apis/axiosClient` folder; because it sets cookies and the base url for us already:

```suggestion
        const response = await axiosInstance.get('/albums/filter',
```

Reply:
These changes have been applied.